server/interpreter.js

```diff
@@ -2048,6 +2047,19 @@ Interpreter.prototype.executeException = function(error) {
   throw realError;
 };
 
+/**
+ * Create and push a new state onto the statestack.
+ * @param {!Object} node AST node for the state.
+ * @return {!Object} New state.
+ */
+Interpreter.prototype.pushNode_ = function(node) {
```

Comment:
Just that in `this.stateStack.pushNode_(node)` is both more concise and clearer than `this.pushNode_(this.StateStack, node)`.

I'd go with option #1.